builtin/pure_ysh.py

```diff
@@ -80,7 +41,7 @@ def Run(self, cmd_val):
             # But should there be a whitelist?
             raise error.Usage('expected a block', loc.Missing)
 
-        pairs = []  # type: List[Tuple[str, value_t]]
+        vars = {}  # type: Dict[str, value_t]
```

Comment:
Oh I think we might need `NewDict()` here

This is a trick/hack to preserve dict ordering in Python (underneath it uses OrderedDict)

In mycpp, our `Dict<K, V>` is defined to be ordered

(I hit this the other day too)